src/components/MagicCodeInput.js

```diff
@@ -215,31 +264,90 @@ function MagicCodeInput(props) {
             }
 
             const newFocusedIndex = Math.max(0, focusedIndex - 1);
+
+            // Saves the input string so that it can compare to the change text
+            // event that will be triggered, this is a workaround for mobile that
+            // triggers the change text on the event after the key press.
+            setInput(TEXT_INPUT_EMPTY_STATE);
+            setFocusedIndex(newFocusedIndex);
+            setEditIndex(newFocusedIndex);
             props.onChangeText(composeToString(numbers));
 
             if (!_.isUndefined(newFocusedIndex)) {
-                inputRefs.current[newFocusedIndex].focus();
+                inputRefs.current.focus();
             }
         }
         if (keyValue === 'ArrowLeft' && !_.isUndefined(focusedIndex)) {
             const newFocusedIndex = Math.max(0, focusedIndex - 1);
-            inputRefs.current[newFocusedIndex].focus();
+            setInput(TEXT_INPUT_EMPTY_STATE);
+            setFocusedIndex(newFocusedIndex);
+            setEditIndex(newFocusedIndex);
+            inputRefs.current.focus();
         } else if (keyValue === 'ArrowRight' && !_.isUndefined(focusedIndex)) {
             const newFocusedIndex = Math.min(focusedIndex + 1, props.maxLength - 1);
-            inputRefs.current[newFocusedIndex].focus();
+            setInput(TEXT_INPUT_EMPTY_STATE);
+            setFocusedIndex(newFocusedIndex);
+            setEditIndex(newFocusedIndex);
+            inputRefs.current.focus();
         } else if (keyValue === 'Enter') {
             // We should prevent users from submitting when it's offline.
             if (props.network.isOffline) {
                 return;
             }
-            setInput('');
+            setInput(TEXT_INPUT_EMPTY_STATE);
             props.onFulfill(props.value);
         }
     };
 
     return (
         <>
             <View style={[styles.magicCodeInputContainer]}>
+                <View style={[StyleSheet.absoluteFillObject, styles.w100, styles.invisibleOverlay]}>
+                    <TapGestureHandler
+                        onBegan={(e) => {
+                            onPress(Math.floor(e.nativeEvent.x / (inputWidth.current / props.maxLength)));
+                        }}
+                    >
+                        {/* Android does not handle touch on invisible Views so I created wrapper around inivisble View just to handle taps */}
```

Comment:
Can you elaborate on this comment?

Why we have it wrapped like this?
```jsx
<View>
      <TapGestureHandler>
             <View> </View>
      </TapGestureHandler>
 </View>
```

Can't we wrap like this instead?
```jsx
      <TapGestureHandler>
             <View> </View>
      </TapGestureHandler>
```

Reply:
bump @wojtus7

Reply:
Okay Keep us posted, thanks!

Reply:
I thing some things changed along the way and I was able to simplify it just to: 

```
<TapGestureHandler>
     <View> // kinda visible only for gesture handler
         <TextInput /> // really invisible to hide input
     </View>
</TapGestureHandler>
```

Reply:
But left the comment why the View between TapGestureHandler and TextInput even exist